src/boat_simulator/boat_simulator/nodes/physics_engine/model.py

```diff
@@ -88,8 +102,71 @@ def __compute_net_force_and_torque(
                 the relative reference frame, expressed in newtons (N), and the second element
                 represents the net torque, expressed in newton-meters (N•m).
         """
-        # TODO Implement this function
-        return (np.array([0, 0, 0]), np.array([0, 0, 0]))
+        # Compute apparent wind and water velocities as ND arrays (2-D)
+
+        assert np.any(rel_wind_vel), "rel_wind_vel cannot be 0"
+        assert np.any(rel_water_vel), "rel_water_vel cannot be 0"
+
+        apparent_wind_vel = np.subtract(rel_wind_vel, self.relative_velocity[0:2])
+        apparent_water_vel = np.subtract(rel_water_vel, self.relative_velocity[0:2])
```

Comment:
Please check if slicing (`[0:2]`) is still necessary after fluid generation update.